AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/CodeGen/Data/DeviceMetadata.swift

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import AWSCognitoIdentityProvider
+import Foundation
+
+enum DeviceMetadata {
+
+    case metadata(Data)
+
+    case noData
```

Comment:
Does it make sense to change this like this:

```suggestion
    case metadata(for: Username, data: Data)

    case noData
```

This way we could always map username with Data and avoid passing empty string like this:

```
.deviceMetadata(.noData, "")
```